hass_nabucasa/events/types.py

```diff
@@ -0,0 +1,48 @@
+"""Event types for cloud system."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from datetime import UTC, datetime
+from enum import StrEnum
+from typing import TYPE_CHECKING, Literal
+
+if TYPE_CHECKING:
+    from ..iot_base import DisconnectReason
+
+
+def _timestamp_factory() -> float:
+    """Generate a timestamp for the current time."""
+    return datetime.now(UTC).timestamp()
+
+
+@dataclass(kw_only=True)
+class _CloudEventBase:
+    """Base class for all cloud events."""
+
+    timestamp: float = field(default_factory=_timestamp_factory)
+
+
+@dataclass(kw_only=True)
```

Comment:
I'd consider making the cloud events frozen. I don't think the consumer should be allowed to change the event.